rust-runtime/aws-smithy-types/test_data/serialize_document.json

```diff
@@ -0,0 +1,42 @@
+{
```

Comment:
> I'm happy to see tests! Testing serialization/deserialization of DynamoDB's [`AttributeValue`](https://docs.rs/aws-sdk-dynamodb/latest/aws_sdk_dynamodb/model/enum.AttributeValue.html) would be nice to see since it's so complex compared to most other generated `enum`s.

It's going to look like this.
```rust
use aws_sdk_dynamodb::{
    model::{
        AttributeValue, ConditionalOperator, ExpectedAttributeValue, ReturnConsumedCapacity,
        ReturnItemCollectionMetrics, ReturnValue,
    },
    operation::PutItem,
};

fn main() {
    // based on the json thing found here https://docs.aws.amazon.com/cli/latest/userguide/cli-services-dynamodb.html
    let putitem = PutItem::builder()
        .table_name("MusicCollection")
        .item("Artists", AttributeValue::S("No One You Know".to_string()))
        .item("SongTitle", AttributeValue::S("Call Me Today".to_string()))
        .item(
            "AlbumTitle",
            AttributeValue::S("Somewhat Famouss".to_string()),
        )
        .return_consumed_capacity(ReturnConsumedCapacity::Total)
        .expected(
            "key",
            ExpectedAttributeValue::builder()
                .exists(true)
                .attribute_value_list(AttributeValue::Bool(true))
                .attribute_value_list(AttributeValue::Null(false))
                .build(),
        )
        .return_values(ReturnValue::AllNew)
        .return_item_collection_metrics(ReturnItemCollectionMetrics::Size)
        .conditional_operator(ConditionalOperator::And)
        .condition_expression("stuff")
        .expression_attribute_names("key", "names")
        .expression_attribute_values(
            "key",
            AttributeValue::Ns(
                [
                    "42.2".to_string(),
                    "-19".to_string(),
                    "7.5".to_string(),
                    "3.14".to_string(),
                ]
                .to_vec(),
            ),
        )
        .build()
        .unwrap();
    let s = serde_json::to_string_pretty(&putitem).unwrap();
    println!("{s}");
}

```
```json
{
  "table_name": "MusicCollection",
  "item": {
    "AlbumTitle": {
      "S": "Somewhat Famouss"
    },
    "Artists": {
      "S": "No One You Know"
    },
    "SongTitle": {
      "S": "Call Me Today"
    }
  },
  "expected": {
    "key": {
      "value": null,
      "exists": true,
      "comparison_operator": null,
      "attribute_value_list": [
        {
          "Bool": true
        },
        {
          "Null": false
        }
      ]
    }
  },
  "return_values": "AllNew",
  "return_consumed_capacity": "Total",
  "return_item_collection_metrics": "Size",
  "conditional_operator": "And",
  "condition_expression": "stuff",
  "expression_attribute_names": {
    "key": "names"
  },
  "expression_attribute_values": {
    "key": {
      "Ns": [
        "42.2",
        "-19",
        "7.5",
        "3.14"
      ]
    }
  }
}
```

Reply:
By the way, how do I add examples/tests that is specific to a certain service?
I couldn't figure out how to do it.